shared/js/url_test/url_test.js

```diff
@@ -0,0 +1,36 @@
+// Tests a URL for response.
+// Use a small image element appended to the body to workaround CORS security.
+function testURL(url, successCallback, failureCallback, timeoutMs) {
+  var img = document.createElement('img');
+  img.width = 1;
+  img.height = 1;
+  img.style['position'] = 'absolute';
+  img.style['bottom'] = 0;
+  img.style['width'] = 1;
+  img.style['height'] = 1;
+  img.style['visibility'] = 'hidden';
+  var called = false;
+  function wrap(callback) {
+    return function() {
+      if (called) {
+        return;
+      }
+      called = true;
+      document.body.removeChild(img);
+      window.clearInterval(interval);
+      window.clearTimeout(timeout);
+      callback();
+    }
+  }
+  var interval = window.setInterval(function() {
+    if(img.complete) {
+      wrap(successCallback)();
+    }
+  }, 250);
+  var timeout = window.setTimeout(wrap(failureCallback), timeoutMs);
```

Comment:
the timeout and interval here are extra fail-safe checks so that eventually the page redirects one way or another. In some tests with Firefox and a throttling proxy, the 'load' callback sometimes wouldn't fire after the image loaded if there was too much of a delay.